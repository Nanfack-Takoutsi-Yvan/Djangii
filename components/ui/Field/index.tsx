/* eslint-disable @typescript-eslint/no-explicit-any */
import { Icon } from "@react-native-material/core"
import {
  StyleProp,
  StyleSheet,
  Text,
  TextInput,
  TextInputProps,
  TextStyle,
  View,
  ViewStyle
} from "react-native"

interface FieldProps extends TextInputProps {
  icon?: string
  label?: string
  value?: string
  iconSize?: number
  iconColor?: string
  placeholder: string
  labelStyle?: StyleProp<TextStyle>
  inputStyle?: StyleProp<TextStyle>
  wrapperStyle?: StyleProp<ViewStyle>
  handleBlur: () => void
  handleChange: () => void
}

export default function Field({
  icon,
  value,
  label,
  iconSize,
  iconColor,
  labelStyle,
  inputStyle,
  placeholder,
  wrapperStyle,
  handleBlur,
  handleChange,
  ...props
}: FieldProps) {
  return (
    <View style={wrapperStyle}>
      {label && <Text style={labelStyle}>{label}</Text>}
      <View style={styles.field}>
        {icon && <Icon name={icon} size={iconSize} color={iconColor} />}
        <TextInput
          {...props}
          value={value}
          style={inputStyle}
          onBlur={handleBlur}
          placeholder={placeholder}
          onChangeText={handleChange}
        />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  field: {
    columnGap: 16,
    alignItems: "center",
    flexDirection: "row",
    backgroundColor: "transparent"
  }
})
