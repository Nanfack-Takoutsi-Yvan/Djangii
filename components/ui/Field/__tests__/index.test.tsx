import * as React from "react"
import renderer from "react-test-renderer"

import Field from "../index"

it(`Field renders correctly`, () => {
  const tree = renderer
    .create(
      <Field
        placeholder="Hello World"
        handleBlur={() => null}
        handleChange={() => null}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
