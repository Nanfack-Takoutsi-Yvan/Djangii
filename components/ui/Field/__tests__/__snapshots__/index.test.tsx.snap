// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "columnGap": 16,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Hello World"
    />
  </View>
</View>
`;
